### ISSUE TITLE
Suggestion: additional association measures for sjt.xtab

### ISSUE THREAD

**Issue by methodds** [sjPlot/devel#187]
Hi,

at the moment ``sjt.xtab`` displays  Phi / Cramer's V for ``show.summary=T``. I'd like to suggest replacing this with a flexible parameter ``corr.method`` similar to ``sjt.correlate``, which would also allow to display either Phi/Cramer's V, Spearman's Rho, or Kendall's Tau (and the corresponding hypothesis test result). 

Social scientists often have categorical variables which they wanna view in crosstables. It would be convenient to have direct access to an appropriate association measure without the need to create a correlation matrix. This would also eliminate the present, somewhat unfortunate situation, where ``show.summary`` is ``TRUE`` by default and in case of ordinal data a mediocre association measure is shown.

**Comment by sjPlot** [comment]
Email-request: the chi square info at the bottom of the table always  displays values using Yates' continuity correction. Is there a way to specify that I don't want to use this correction?

**Comment by sjPlot** [comment]
What association measure would be appropriate for which type of variables? Would it be enough to check for this type (numeric, factor, ordered...) or would you suggest specifying the measure as argument to sjt.xtab?

**Comment by methodds** [comment]
I would recommend to let the analyst choose as an argument. There are like 2304982340 measures but very common are nominal: Cramer's V, ordinal: Kendall's Tau and Spearman, cardinal: Pearon's R.

**Comment by methodds** [comment]
Tau Spearman and Pearsons are all available in Base R's correlation functions. :)

**Comment by methodds** [comment]
I'dbe super awesome if p values would also be displayed for the corresponding measure.

**Comment by sjPlot** [comment]
And Phi beside Cramer's V, for 2x2?

**Comment by sjPlot** [comment]
For p-value: `chisq.test()`, `fisher.test()`, and `cor.test()`? What about McNemar?

**Comment by methodds** [comment]
Sure why not. To be honest though I never needed one specifically for 2x2 tables in my life.

**Comment by methodds** [comment]
> For p-value: chisq.test(), fisher.test(), and cor.test()? What about McNemar?

```chisq.test()``` and ```cor.test()``` together would cover Cramer's V, Spearman, Kenndall's Tau and Pearson's R. I'm not familiar with McNemar sorry.

**Comment by sjPlot** [comment]
Which of the return values from `cor.test()` is the required statistics to be reported? `statistic` or `estimate`?

**Comment by methodds** [comment]
````
> x <- c(44.4, 45.9, 41.9, 53.3, 44.7, 44.1, 50.7, 45.2, 60.1)
> y <- c( 2.6,  3.1,  2.5,  5.0,  3.6,  4.0,  5.2,  2.8,  3.8)
> 
> cor(x,y, method='kendall')
[1] 0.4444444
> 
> test <- cor.test(x, y, method = "kendall")
> test$p.value
[1] 0.1194389
> test$estimate
      tau 
0.4444444 
> test$statistic
 T 
26 
````
``estimate`` is the result of the correlation test and ```statistic``` is the value of the corresponding distribution used for the test (in this case the [t-distribution](https://en.wikipedia.org/wiki/Student's_t-distribution)).

**Comment by methodds** [comment]
Sorry that didn't really answer your question. Actually everything could be returned, although I'm not sure whether this takes too much space? What do you think ? Return correlation value, test statistic (least important) and p-value?

**Comment by methodds** [comment]
And just to be precise: ```cor.test(method='kendall') returns [tau b](https://en.wikipedia.org/wiki/Kendall_rank_correlation_coefficient#Tau-b), which is totally fine.

**Comment by sjPlot** [comment]
Ok, [this commit](https://github.com/sjPlot/sjstats/commit/1478d2fbf45442b6936f653acfa3b6973003b6be) addresses this issue. However, due to mutual dependencies of packages, I have to submit to CRAN stepwise, i.e. it'll take some time until this issue is close.

I need to submit sjPlot first, so I can submit the updates of sjmisc (the package I actually want to submit). Then I need to submit sjPlot again later (which will use the `std()`- and `center()`-functions that were moved from _sjstats_ to _sjmisc_), after that I can submit _sjstats_, and then I could submit sjPlot again, which then calls `sjstats::xtab_statistics()`...

Or maybe I'll just implement that function as hidden in the sjPlot package, might be faster... ;-)

**Comment by methodds** [comment]
Thanks for the updates :). Regarding new CRAN releases, would you consider the state of your packages somewhat stable, meaning that the core functions are unlikely to change damatically in the near future? 
I'll use the packages in April for teaching, that's why I'm asking.

**Comment by sjPlot** [comment]
sjPlot is "stable", except for possible bugs I'm not yet aware of. the forthcoming update of sjmisc means some major changes according to the api / function arguments, but that would be the current way I'm planning to go (see vignette "design philosophy"), so it's less likely to change big in future updates (except new functions of course). sjstats is also "stable", except that `std()` and `center()` will be removed soon, because these methods are now in sjmisc.

So, in short: the basic api / function design of my packages won't change substantial in the future and the design is considered as "stable".

**Comment by sjPlot** [comment]
btw, I'm using my packages for teaching, beginning in march - that's why I'm trying to revise my packages and made some thoughts about the "design philosophy" behind the api and functions...
